### PR TITLE
[move] Remove useless/unused/legacy code

### PR DESCRIPTION
**Pull request by georgemitenkov**
### Description

This PR removes legacy Move code. In particular:

1. `std::vector` natives are removed, because bytecode instructions are used.
2. `vault`/`events` are removed because they are not really useful and `events` exist in Aptos repo. If we need events,
    that code can be placed instead!
3. `std::debug` is removed. We have debug functionality at Aptos already, and users can (and should) be using it already.
    We can make it available to Move if there is demand, otherwise no point of keeping old code.

Legacy abstract value size computations are removed because 1) gas meter is used for testing and so they are not important; 2) it is up to clients to decide on these APIs and values should not know about it.

### CODE REVIEW

File: third_party/move/move-vm/types/src/values/value_prop_tests.rs [hunk 1]
```diff
@@ -12,13 +11,5 @@ proptest! {
         let blob = value.simple_serialize(&layout).expect("must serialize");
         let value_deserialized = Value::simple_deserialize(&blob, &layout).expect("must deserialize");
         assert!(value.equals(&value_deserialized).unwrap());
-
-        let move_value = value.as_move_value(&layout);
```
Comment (runtian-zhou) [inline third_party/move/move-vm/types/src/values/value_prop_tests.rs:16]: Why are we removing this?

File: third_party/move/tools/move-cli/tests/sandbox_tests/print_stack_trace/scripts/print_stack_trace.move [hunk 2]
```diff
@@ -1,16 +0,0 @@
-script {
-use std::debug;
-use std::vector;
-use 0x2::N;
-
-fun print_stack_trace() {
```
Comment (runtian-zhou) [inline third_party/move/tools/move-cli/tests/sandbox_tests/print_stack_trace/scripts/print_stack_trace.move:6]: Same for those tests?

File: third_party/move/move-stdlib/src/natives/debug.rs [hunk 3]
```diff
@@ -1,652 +0,0 @@
...
-            "print_stack_trace",
-            make_native_print_stack_trace(gas_params.print_stack_trace),
-        ),
-    ];
-
-    make_module_natives(natives)
-}
-
-#[cfg(feature = "testing")]
-mod testing {
```
Comment (runtian-zhou) [inline third_party/move/move-stdlib/src/natives/debug.rs:145]: Was wondering if we should remove those debug natives. They could be used by some external folks?

Comment (georgemitenkov) [reply third_party/move/move-stdlib/src/natives/debug.rs:145]: Can't external folks use aptos debug natives? Do we know if they use them? If not I would prefer to get rid of these and shift our debug implementation to Move repo for example :)

Comment (runtian-zhou) [reply third_party/move/move-stdlib/src/natives/debug.rs:145]: cc @jolestar

File: third_party/move/move-vm/types/src/values/value_prop_tests.rs (hunk 1 above)
Comment (georgemitenkov) [reply third_party/move/move-vm/types/src/values/value_prop_tests.rs:16]: As far as I understand, we have `Value` for runtime and `MoveValue` used as a core type (not sure what for but its more stable I was told before I think). Here the idea was to have an identical value in 2 representations and check it is the same, maybe I should have kept this :)

Comment (georgemitenkov) [reply third_party/move/move-vm/types/src/values/value_prop_tests.rs:16]: Let me make `as_move_value` private and test only. The code there is a bit ugly, so many panics....

File: third_party/move/move-stdlib/src/natives/debug.rs (hunk 3 above)
Comment (jolestar) [reply third_party/move/move-stdlib/src/natives/debug.rs:145]: We use `std::debug`, and many examples docs use `std::debug`. I suggest we keep the `std::debug` or shift `aptos_std::debug` to `std::debug`.

Comment (georgemitenkov) [reply third_party/move/move-stdlib/src/natives/debug.rs:145]: Good to know! Thanks!

Comment (vgao1996) [reply third_party/move/move-stdlib/src/natives/debug.rs:145]: @georgemitenkov I don't think external folks can use the latest version of Aptos natives easily -- those are now dependent on the our gas schedule.

Comment (georgemitenkov) [reply third_party/move/move-stdlib/src/natives/debug.rs:145]: Makes sense, closing the PR and let's plan more incremental changes in this case.

**Pull request closed without merging**